### ISSUE TITLE
[BUG] Key repeat does not work while using ibus-typing-booster

### ISSUE THREAD

**Issue by pallaswept** [mike-fabian/ibus-typing-booster#475]
**Describe the bug**
Key repeat does not work while using this tool. By this I mean,  let's say I hold down the period ('.') key, normally there is a brief pause, and then the key will repeat and add more '.' until I release the key. 
This is handy when typing ellipses for example, or if I'm being loose with my language and want to type something like "yessssssssss!" 

**To Reproduce**
1. Enable ibus-typing-booster as usual.
2. Hold down any key which would normally repeat.
3. Note that only a single character is added to the preedit regardless of how long the key is held down.

**Expected behavior**
Key will repeat if held down

**ibus-typing-booster version?**
2.24.10

**ibus version?**
IBus 1.5.29-rc2

**Distribution and version?**
OpenSUSE Tumbleweed

**Desktop and version?**
KDE 5.27

**Xorg or Wayland?**
Wayland

**Additional context**
I'm not sure if this is actually a problem with ibus-typing-booster or ibus or KDE's virtual keyboard implementation. Apologies if this should have been filed upstream somewhere,  I'm very new to this whole virtual keyboard ecosystem.

**Comment by mike-fabian** [comment]
Are you using a virtual keyboard or a hardware keyboard?

**Comment by pallaswept** [comment]
Physically, a hardware keyboard, but then I select a virtual keyboard in KDE to enable this tool:
![image](https://github.com/mike-fabian/ibus-typing-booster/assets/132128962/0aafe379-650c-44b8-ae77-398fbd0cbdd0)

**Comment by mike-fabian** [comment]
I can reproduce this, but it has nothing to do with typing booster, happens always when ibus is active in KDE Plasma Wayland.

**Comment by pallaswept** [comment]
Thanks for looking into it for me, sounds like I need to file this upstream. Do you know if I should be filing it under ibus or KDE?

I do have IceWM (X11) and I can try a KDE X11 session, that will give me a hint, I guess... I'll let you know how I fare.

**Comment by mike-fabian** [comment]
This video shows a Plasma Wayland session. I am using ibus-anthy instead of ibus-typing-booster. It does not matter which ibus input method is used, it happens with all of them, not only with ibus-typing-booster. So it is not an ibus-typing-booster problem.

In the video I type into

- konsole:  autorepeat works
- kwrite:   autorepeat works
- The dialog to start a command (Alt+F2):     autorepeat works
- gedit:   autorepeat does **not** work
- gnome-terminal:    autorepeat does **not** work

I always press the `a` key which gives me the hiragana character あ which is then repeated when autorepeat works.

In gedit and gnome-terminal, where it does not work, I release the key and try again several times.

It behaves the same with the backspace key, in konsole, kwrite, and Alt+F2,  the autorepeat for backspace works, in gedit and gnome-terminal it does **not** work.


https://github.com/mike-fabian/ibus-typing-booster/assets/2330175/d12eefbc-b29e-478d-90d0-c51ba0d53276

**Comment by mike-fabian** [comment]
Here I do the same in a Plasma X11 session. No problem, key repeat works everywhere:

https://github.com/mike-fabian/ibus-typing-booster/assets/2330175/f0bb7705-5a7d-4501-ba6e-1386f60db88b

**Comment by pallaswept** [comment]
> So it is not an ibus-typing-booster problem.

Definitely not,  that is why I closed this issue already :) I'm sorry I filed it in the wrong place! I don't have any other ibus tools or DE's which I use, so there was no way for me to realise that this was the wrong place, but I understand now. It's very kind of you to help me to troubleshoot this issue which is clearly not even your responsibility, so thank you very much :)

Strangely,  I cannot reproduce your results. For me, when I press a key, it does not appear in the preedit at all until I release the key, regardless of using a Qt or GTK app - so there is no chance of a character repeat, there is not even one character appearing until I release the key - but just as I was typing that, now I see your last video, I think that this might be related to the keyboard layout, because I am using the US English layout and it behaves just like that. But I tried anthy and some other layouts and it was always like this.  I'm not sure. 

I am sure, this is not a problem with your application though, so I thank you for your help 🙂
